src/Bladeburner/Actions/TeamCasualties.ts

```diff
@@ -26,22 +26,31 @@ export interface TeamActionWithCasualties {
  * and may result in casualties, reducing the player's hp, killing team members
  * and killing sleeves (to shock them, sleeves are immortal) *
  */
-export function resolveTeamCasualties(action: TeamActionWithCasualties, team: OperationTeam, success: boolean) {
-  const severity = success ? CasualtyFactor.LOW_CASUALTIES : CasualtyFactor.HIGH_CASUALTIES;
-  const radius = action.teamCount * severity;
-  const worstCase = severity < 1 ? Math.ceil(radius) : Math.floor(radius);
-  /** Best case is always no deaths */
-  const deaths = team.getTeamCasualtiesRoll(action.getMinimumCasualties(), worstCase);
-  const humans = action.teamCount - team.sleeveSize;
-  const humanDeaths = Math.min(humans, deaths);
-  /** Supporting Sleeves take damage when they are part of losses,
-   *   e.g. 8 sleeves + 3 team members with 4 losses -> 1 sleeve takes damage */
-  team.killRandomSupportingSleeves(deaths - humanDeaths);
-
-  /** Clamped, bugfix for PR#1659
-   * "BUGFIX: Wrong team size when all team members die in Bladeburner's action" */
-  team.teamSize = Math.max(team.teamSize - humanDeaths, team.sleeveSize);
-  team.teamLost += deaths;
-
-  return deaths;
+export function resolveTeamCasualties(action: TeamActionWithCasualties, team: OperationTeam, success: boolean): number {
```

Comment:
The refactor revert is over-engineered. Here is the minimal fix needed by following your clamp suggestions in the PR description (insert at pre-change Line number 34):

` 
  const bestCase = Math.max(0, Math.min(action.teamCount, action.getMinimumCasualties()));
  const deaths = team.getTeamCasualtiesRoll(bestCase, bestCase < worstCase ? worstCase : bestCase);
  const humans = Math.max(0, action.teamCount - team.sleeveSize);
`

Reply:
Unless @d0sboots demands, I won't change the implementation of this PR. It uses the old logic and looks very similar to the old code. That's the whole point of the fix.

I also disagree about the "over-engineered" part. For example:
```js
const severity = success ? CasualtyFactor.LOW_CASUALTIES : CasualtyFactor.HIGH_CASUALTIES;
const radius = action.teamCount * severity;
const worstCase = severity < 1 ? Math.ceil(radius) : Math.floor(radius);
```
vs
```js
const maxCasualties = success
    ? Math.ceil(action.teamCount * CasualtyFactor.LOW_CASUALTIES)
    : Math.floor(action.teamCount * CasualtyFactor.HIGH_CASUALTIES);
```
The latter code is how it was implemented in the past, and it looks cleaner.

Reply:
Your focus on cleanliness I agree with. Let's work with the data: 6 different people missed this bug after 3 separate PRs over 4 weeks. I invite you to use neither this implementation nor the old one if you want to focus on making it more readable.

But also let's focus on making small improvements in the same vein as you suggested your comments of the original PR: either fix the bug or refactor it, but not both.

Personally I prefer the math problem being laid out flat, rather than resulting to control structures and early returns. But that's just my preference (and bias), no need to use my version. However, I invite you not to give into the fear-based option of "the old version worked fine as long as you don't touch it".

If you think the refactor was a fluke that all the reviewers, yourself included, missed then revert that commit instead of gradually reintroducing the old confusion.

Reply:
To make sure that this discussion does not go awry, I'll make this clear:
- I have no hard feelings about the refactor. Please check the last paragraph. Your refactor is a good addition/change.
- I don't want to change the implementation, but it's **not** because I think my code is "better" or anything. I choose to stick with the old implementation because it's bug-free (AFAIK) while still being easy to understand (this is an opinion, not a fact). Please note that I don't have any implication when saying that.

"X people do not find problems for Y <PRs/weeks/months/years>": I have seen many people use this phrase in their arguments in many cases, and it never makes sense to me. One example: OpenSSL, in spite of being an extremely important project and being "reviewed" by innumerable experts, still has nasty bugs. Back in the "era" of Heartbleed, many people asked: How has this bug not been found out when <Insert tons of reason here> before it's merged? To be honest, I was not surprised at all. "Having many eyes on it" is not the silver bullet.

I know that comparing Bitburner with OpenSSL (or other massive projects) is silly (to make this message succinct, I won't write a useless wall of text about it). The point is that we all make mistakes. We try to minimize the chance of having bugs by writing test cases, keeping the code clean and easy to read, etc. (Please note that I don't imply that your code is buggy or anything, just in case you misunderstand my point). These bugs, which are outlined in this PR, are caused by the new logic, so I reuse the old logic to fix it (more about this in the next paragraph). I pointed out the potential problem of the new logic in https://github.com/bitburner-official/bitburner-src/pull/1654#discussion_r1763441473, but I did not check it again after that, before d0sboots merged it. If there is a person to blame, it's me. There is no need to pull other people in. It's my fault for not checking it again before it's merged, and I'll take the blame for it.

This PR is a bugfix one, and I do that by reusing the old code. The old implementation is clean and easy to understand, so I decided to reuse it, not because "the old version worked fine as long as you don't touch it". This is just my opinion, not a fact, so it's fine if you disagree. We have made our points clear, so I don't think we should argue about which one is "better". I'll let the maintainer decide the implementation. One small point: I did not write those old code, so I have no special "feelings" about them. I'm fine if the maintainer tells me to use X code instead of Y code.

The last point: your refactor is not a "fluke". I have never said or thought about it like that. I have also never said that I would revert it or implement another "better" version. I spotted some bugs, then I made a PR to fix them. No hard feelings. In the past, I have also introduced some nasty bugs in this game (corrupted save data in the "Save compression" PR, "unlimited" SF in the "BN option" PR, etc.) when I submitted my PRs. We spot bugs, we fix them, then we move on to other bugs/suggestions.

Reply:
I don't mind either way, just sad to see the old control flow reintroduced as it made (in my opinion) the sleeved parts much harder to follow.

Reply:
Random points:
* When a bug is found in a refactor, a revert (or partial revert) should generally be the go-to response. So in that sense, what CatLover is doing makes perfect sense to me. Especially since our tests are... not the greatest.
* I don't have a particular opinion about either way being easier to read.
* Kudos for keeping this focused on the code.